### PR TITLE
Allow forcing clusterchecks rebalancing for scheduler workers utilisation rebalancing  algo

### PR DESCRIPTION
**Pull request by adel121**
### What does this PR do?

Allow forcing clusterchecks rebalancing for scheduler workers utilisation rebalancing  algorithm

### Motivation

There’s a command to execute a cluster checks rebalance: `datadog-cluster-agent clusterchecks rebalance`

The code doesn’t always rebalance, it only does so if the new checks distribution that it found is better than the current one by a threshold defined. We do this to avoid re-scheduling too many checks for little benefit.

This PR introduces a new command `datadog-cluster-agent clusterchecks rebalance --force` that skips those checks. This could be used as a quick solution when we detect an issue. The current solution is to kill the leader DCA manually.

### Additional Notes
- Forcing a rebalance doesn't necessarily mean that cluster checks dispatching will be changed. This is because it is possible that the new proposed distribution is identical to the current one, and thus no clusterchecks are rescheduled.
- We have 2 possible algorithms to rebalance clusterchecks. One of them uses busyness to decide whether (and how) to rebalance the checks. The second one uses the scheduler workers utilisation. Forcing rebalancing is being enabled only for the latter.
- Using the `--force` option while not using rebalancing with utilisation will not result in forcing rebalancing.

### Describe how to test/QA your changes

- Deploy the agent and cluster agent enabling the clustercheck runners and the cluster agent. You should set the environment variables (see sample below) to allow advanced dispatching and rebalancing with utilisation. Set the replicas of the clusterchecks runners to 1. Here is a sample configuration using helm:
```
clusterChecksRunner:
  enabled: true
  replicas: 1
  image:
    repository: datadog/agent
    tag: 7.48.0-rc.9-jmx
    doNotCheckTag: true

clusterAgent:
  enabled: true

  env:
    - name: DD_CLUSTER_CHECKS_ADVANCED_DISPATCHING_ENABLED
      value: true
    - name: DD_CLUSTER_CHECKS_REBALANCE_WITH_UTILIZATION
      value: true

  image:
    repository: clusteragent-image
    tag: latest
    doNotCheckTag: true
    pullPolicy: Always
```
- Install the helm chart on a 3 node minikube cluster locally (or on any kubernetes cluster)
- Make sure all agent, cluster agent, and clustercheck runners are running
- Create 2 services with autodiscovery. To use autodiscovery you can add annotations to the service as such:
```
annotations:
    ad.datadoghq.com/service.checks: |
      {
        "http_check": {
          "init_config": {},
          "instances": [
            {
              "url":"http://%%host%%",
              "name":"My Nginx",
              "timeout":2
            }
          ]
        }
      }
```
- You can then create pods matching the services' selectors and annotations. (For the annotation above, you can create a pod with `nginx` as an image, and give it a label that is identical to the `selector` set in the service)
- This should configure 2 cluster checks and dispatch them to the single replica of the clustercheck runners that exist on the cluster. (you can verify this by checking the logs on the cluster agent pod using `kubectl logs <pod-id>`, this should show messages regarding dispatching both cluster checks on the clusterchecks runner)
- Use `kubectl edit deployment` to increase the number of replicas of the cluster checks runners to 3.
- You can then try rebalancing the clusterchecks by running the following command on the cluster agent pod: `datadog-cluster-agent clusterchecks rebalance --force` 
- Executing the same command several times should force rebalancing most of the times. (NOTE: it is possible that sometimes no clusterchecks are not rebalanced, which is expected)
- You can then check the logs on the cluster agent and verify messages saying that rebalancing was being forced:  **Forcing rebalance proposed distribution for the cluster checks. Utilization stdDev of proposed distribution: 0.000. StdDev of current distribution: 0.000. Proposed distribution: ....**

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes-dca/notes/allow-force-clusterchecks-rebalancing-46b8f58eb157ecde.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG-DCA.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Allow forcing clusterchecks rebalancing with utilisation.
```
[the remark this answers is not in the document]
Comment (alai97) [reply releasenotes-dca/notes/allow-force-clusterchecks-rebalancing-46b8f58eb157ecde.yaml:11]:
```suggestion
    Added a new --force option to the `datadog-cluster-agent clusterchecks rebalance` command that allows you to force clustercheck rebalancing with utilization.
```

**Review by alai97 — APPROVED** [review]
Looks good!

**Review by davidor — APPROVED** [review]

File: pkg/cli/subcommands/clusterchecks/command.go [hunk 2]
```diff
@@ -72,12 +73,15 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
...
 
 			return fxutil.OneShot(rebalance,
+				fx.Supply(cliParams),
 				fx.Supply(bundleParams(globalParams)),
 				core.Bundle,
 			)
 		},
 	}
 
+	rebalanceCmd.Flags().BoolVarP(&cliParams.force, "force", "f", false, "Use to force rebalance")
```
Comment (sgnn7) [inline pkg/cli/subcommands/clusterchecks/command.go:83]: This seems like it could use a test in `command_test`

**Review by sgnn7 — APPROVED** [review]
Just need a test or two but LGTM for ASC files

**Pull request merged**